Ruby/wc/ccwc.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require_relative './file_content'
+
+def find_input_command(input_array)
+  return '-c' if input_array.include?('-c')
+  return '-l' if input_array.include?('-l')
+  return '-w' if input_array.include?('-w')
+  return '-m' if input_array.include?('-m')
+
+  'invalid'
+end
+
+def print_info(count, file_name)
+  puts "#{count} #{file_name}" unless file_name.nil?
+  puts count if file_name.nil?
+end
+
+def find_index_of_file_and_command(input_array)
+  input = find_input_command(input_array[0])
+  return [input, 1] if input != 'invalid'
+
+  input = find_input_command(input_array[1])
+  [input, 0] if input != 'invalid'
+end
+
+def read(file)
+  file_content = ''
+  File.foreach(file) do |line|
+    file_content += line
+  end
+  file_content
+end
+
+input = ''
+# `ARGV` is a special global variable that holds the command-line arguments passed to a Ruby script.
+input_array = ARGV
+command_index = find_index_of_file_and_command(input_array)
+if !$stdin.tty?
+  # STDIN.tty? to check if the standard input is interactively coming from the terminal.
+  # eg: `cat test.txt | ruby ccwb.rb -c`
+  # in the above input - test.txt is not being directly used as the standard input to the ruby script,
+  # rather the contents of the file are used an input.
+  #
+  # `$stdin` is a global variable that represents the standard input stream.
+  # The use of $stdin in this context allows the script to read from the standard input,
+  # which is being provided through the pipe (|) from the cat command.
+  file_content = ''
+  $stdin.each_line do |line|
+    file_content += line
+  end
+  input = command_index[0]
+elsif input_array.length > 1
+  input = command_index[0]
+  file_name = input_array[command_index[1]]
+  file_content = read(file_name)
+elsif input_array.length.positive?
+  file_name = input_array[0]
+  file_content = read(file_name)
+end
```

Comment:
NIT: We're reading the entire file into memory then wrapping it in our FileContent object - I'm guessing if the file is too big then it'd be a problem.
Would this work? https://www.rubydoc.info/stdlib/core/Enumerator/Lazy 
To stream the input instead of reading all of it into memory; just a thought 🤔

Reply:
I'm reading the file here line by line - https://github.com/aganapuram/CodingChallengesJC/pull/1/files#diff-84e265f1412bfd60c646a99e7ebfea2fee7db76069b1cdd9009c2dca1a8d178bR27

wouldn't this stream the file content?

Reply:
Sorry - worded it in a poor way.
So the case that came to head were what if the file size is really really big - so generally what I've seen is we can switch from `map` like operation - read entire list, then apply calculation on the entire list 
to `stream` like operation - apply calculation on each element of list and accumulate the result as we go. 

From what I can tell this program's memory footprint will grow according to the input supplied.

Per req, we'd need to read till EOF but there's no reason to avoid it but there's no reason to hold entire content - 
start with initial value of 0,
read a line then add result of prev calc to  word count / line count / character count 
when EOF - print out results 

Would this work?